WordPress/Classes/ViewRelated/NUX/SigninHelpers.swift

```diff
@@ -0,0 +1,275 @@
+import UIKit
+import NSURL_IDN
+import WordPressComAnalytics
+
+/// A collection of helper methods for NUX.
+///
+@objc class SigninHelpers: NSObject
+{
+    private static let AuthenticationEmailKey = "AuthenticationEmailKey"
+
+    /// The base site URL path derived from `loginFields.siteUrl`
+    ///
+    /// - Parameters:
+    ///     - string: The source URL as a string.
+    ///
+    /// - Returns: The base URL or an empty string.
+    ///
+    class func baseSiteURL(string: String) -> String {
+        guard let siteURL = NSURL(string: NSURL.IDNDecodedURL(string)) else {
+            return ""
+        }
+
+        var path = siteURL.absoluteString.lowercaseString
+
+        if path.isWordPressComPath() {
+            if siteURL.scheme.characters.count == 0 {
+                path = "https://\(path)"
+            } else if path.rangeOfString("http://") != nil {
+                path = path.stringByReplacingOccurrencesOfString("http://", withString: "https://")
+            }
+        } else if siteURL.scheme.characters.count == 0 {
+            path = "http://\(path)"
+        }
+
+        let wpLogin = try! NSRegularExpression(pattern: "/wp-login.php$", options: .CaseInsensitive)
+        let wpadmin = try! NSRegularExpression(pattern: "/wp-admin/?$", options: .CaseInsensitive)
+        let trailingSlash = try! NSRegularExpression(pattern: "/?$", options: .CaseInsensitive)
+
+        path = wpLogin.stringByReplacingMatchesInString(path, options: .ReportCompletion, range: NSRange(location: 0, length: path.characters.count), withTemplate: "")
+        path = wpadmin.stringByReplacingMatchesInString(path, options: .ReportCompletion, range: NSRange(location: 0, length: path.characters.count), withTemplate: "")
+        path = trailingSlash.stringByReplacingMatchesInString(path, options: .ReportCompletion, range: NSRange(location: 0, length: path.characters.count), withTemplate: "")
+
+        return path
+    }
+
+
+    // MARK: - Validation Helpers
+
+
+    /// Checks if the passed string matches a reserved username.
+    /// 
+    /// - Parameters:
+    ///     - username: The username to test.
+    ///
+    class func isUsernameReserved(username: String) -> Bool {
+        let name = username.lowercaseString.trim()
+        return ["admin", "administrator", "root"].contains(name)
```

Comment:
A fair point but since I'd likely still initialize the set with an array literal I'll leave it as is for more compact code.